### ISSUE TITLE
JVM SIGSEGV crash on Linux (Debian, Ubuntu, CentOS, Alpine)

### ISSUE THREAD

**Issue by dmumpuu** [duckdb/duckdb#8931]
### What happens?

The JVM crashes on Linux when trying to `COPY` to S3 (AWS S3, LocalStack). The issue doesn't happen on MacOS/Windows.
Tried on OpenJDK, Temurin, Liberica, versions: 17, 19, 20

<details>
  <summary>Dump</summary>

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f54182d2194, pid=26566, tid=26568
#
# JRE version: OpenJDK Runtime Environment Temurin-20.0.2+9 (20.0.2+9) (build 20.0.2+9)
# Java VM: OpenJDK 64-Bit Server VM Temurin-20.0.2+9 (20.0.2+9, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, linux-amd64)
# Problematic frame:
# C  [libstdc++.so.6+0xd2194]  std::__codecvt_utf8_utf16_base<char16_t>::do_unshift(__mbstate_t&, char*, char*, char*&) const+0x4
#
# Core dump will be written. Default location: Core dumps may be processed with "/usr/share/apport/apport -p%p -s%s -c%c -d%d -P%P -u%u -g%g -- %E" (or dumping to /path/to/file.Projects/untitled/core.26566)
#
# If you would like to submit a bug report, please visit:
#   https://github.com/adoptium/adoptium-support/issues
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#

---------------  S U M M A R Y ------------


Host: AMD Ryzen 5 5600H with Radeon Graphics, 12 cores, 14G, Ubuntu 22.04.3 LTS
Time: Thu Sep 14 15:20:53 2023 CEST elapsed time: 14.488186 seconds (0d 0h 0m 14s)

---------------  T H R E A D  ---------------

Current thread (0x00007f544402d2b0):  JavaThread "main" [_thread_in_native, id=26568, stack(0x00007f544b100000,0x00007f544b200000)]

Stack: [0x00007f544b100000,0x00007f544b200000],  sp=0x00007f544b1fd698,  free space=1013k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
C  [libstdc++.so.6+0xd2194]  std::__codecvt_utf8_utf16_base<char16_t>::do_unshift(__mbstate_t&, char*, char*, char*&) const+0x4
Java frames: (J=compiled Java code, j=interpreted, Vv=VM code)
j  org.duckdb.DuckDBNative.duckdb_jdbc_execute(Ljava/nio/ByteBuffer;[Ljava/lang/Object;)Ljava/nio/ByteBuffer;+0
j  org.duckdb.DuckDBPreparedStatement.execute()Z+67
j  org.duckdb.DuckDBPreparedStatement.execute(Ljava/lang/String;)Z+6
j  MainKt.main()V+417
j  MainKt.main([Ljava/lang/String;)V+0
v  ~StubRoutines::call_stub 0x00007f5434137cc6

siginfo: si_signo: 11 (SIGSEGV), si_code: 1 (SEGV_MAPERR), si_addr: 0x0000000000000020
. . . 
```

</details>

Probably this issue is related to https://github.com/duckdb/duckdb/issues/7596

### To Reproduce

```java
stmt.execute("CREATE TABLE tbl1 (int0 INTEGER, int1 INTEGER)")
stmt.execute("INSERT INTO tbl1 VALUES (1,2), (3,4)")
stmt.execute("COPY tbl1 TO 's3://test-bucket/tbl1' (PARTITION_BY (int0, int1), FORMAT PARQUET, CODEC ZSTD, OVERWRITE_OR_IGNORE 1)")
```

### OS:

Linux (Debian, Ubuntu, CentOS, Alpine), Docker

### DuckDB Version:

0.8.1 (0.7.1 as well)

### DuckDB Client:

Java Client

### Full Name:

Dmitry Popov

### Affiliation:

self-employed

### Have you tried this on the latest `main` branch?

I have tested with a main build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by Mause** [comment]
Hi there, you selected that you tested with a 'main' (dev) version - which version did you test with? We recently merged a fix for these issues

**Comment by dmumpuu** [comment]
> Hi there, you selected that you tested with a 'main' (dev) version - which version did you test with? We recently merged a fix for these issues (#8685)

Sorry, seems I misunderstood the options - I tested with the 'release' versions: 0.8.1 and 0.7.1.
I'm not sure that exception handling (https://github.com/duckdb/duckdb/pull/8685) will help with the issue.
But I've seen a couple of fixes avoiding static linking with `libc/libc++` - hope that's the proper way)

P.S. I tried with snapshot version (0.9.0-20230914.014458-349) - no SIGSEGV appeared
